.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,55 @@
+name: PR validation
+
+on:
+  push:
+    branches: [master, dev]
+  pull_request:
+    branches:
+    - dev
+    - master
+
+jobs:
+  dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: /home/runner/work/hathor-explorer/hathor-explorer/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install Dependencies
+        run: |
+          npm install
+          tar -cvf node_modules.tar ./node_modules
+      - name: Upload node modules
+        uses: actions/upload-artifact@v2
```

Comment:
It doesn't seem necessary to upload artifacts in this case, it will just make things slower.

We could just remove this and bring the `Build` step below here.

```
      - name: Install Dependencies
        run: |
          npm install
      - name: Build
        run: |
          make testnet_build
```

Reply:
Makes sense, much better. Did some changes in the file, can you take a look again please?